src/main/io/piniobox.h

```diff
@@ -22,3 +22,4 @@
 
 void pinioBoxInit(const pinioBoxConfig_t *pinioBoxConfig);
 void pinioBoxUpdate(timeUs_t currentTimeUs);
+void pinioBoxTaskControl(void);
```

Comment:
This is a good find, and I even think this is a bug affecting the performance, so it should be considered for 4.2. With this faster loop times might be possible / more stable on some systems that do not use pinio, which will include most F411 boards.

But I am not sure that moving the `setTaskEnabled()` into the unit defining the functionality is the right move - `setTaskEnabled()` is so far treated as an internal to `tasks.c`, and I think having a `bool pinioBoxEnabled(void)` in `piniobox.c` and enabling the task in `tasks.c` will fit the existing pattern better.

Reply:
I did it this way because I originally went further and checked whether any of the linked box ID’s had valid activation ranges. This then needed to enable/disable in the MSP and CLI as the `aux` entries were updated. Decided that was too much and not worth the gain.

However left it as an accessible function because I figured we will probably expose pinio in the Configurator at some point and need to call it from msp.c then.

That’s why the function can disable as well. For the existing task startup case for only enabling is needed.

Reply:
Ok.

To me, the firmware configuration can be separated into two distinct parts:
1. hardware configuration that controls what hardware feature a board has. Pre-defined by the target maintainer, and not normally exposed to the user in the GUI, as changing it will normally result in the board not performing as specified. Only guaranteed to apply after a reboot;
2. craft configuration. Exposed to the user in the GUI. Configuration should apply without a reboot.

For me, `pinio_box` and `pinio_config` falls into 1., so there should never be a need to change this through MSP, or any other way, at runtime.

Reply:
`pinio_box` links a pin to a mode. That’s something that could clearly be an extension to the Modes tab. Or from the reverse based on an interface by pin (settings and box link). Nobody knows what `pinio_box = 40,255,255,255` means or that “40” is USER1 or what number any mode is. So having to link them in the CLI means that users have no idea how to use it short of supplied instructions.

Reply:
Do you have any examples of boards that come with pre-defined pinio pins where the user is expected to link these pins to a mode themselves?

Reply:
In my opinion this is kind of backwards logic. We should be making the pinio functionality (which is very powerful) more accessible. Not expect that people will only use what the manufacturer set up in their target. And before boards started coming with on-board vtx switches for example people were rolling their own with mosfets which then evolved into RealPit, etc. Had another example on FB where somebody wanted to control a beeper directly with a switch (constant tone - just on/off). Yet another case was somebody wanting to do a "candy drop" for a birthday party and use pinio to drive a mosfet/solenoid. The point is that pinio is very powerful, but also very obscure and not end-user friendly.

Reply:
Not sure about that - to me, GUI configuration should be added for functionality that a significant percentage of users will want to change. Adding configuration like pin assignments to the GUI will only make configurator harder to use for the average user, as there will be things that will cause the flight controller to stop functioning if they change them without having the knowledge of how to do it in the right way.
We now have ~40 targets that come with one or more pins preconfigured for pinio use (mostly VTX switch). The average user will be able to use this safely from the Modes tab with no need to ever change `pinio_config` or `pinio_box`. With #9578 manufacturers can even add appropriate labels for custom pinio modes implemented on their boards. If we add the facility to change the pinio-to-mode configuration in the GUI this just adds a way for the average user of these boards to change the pre-configured setup and stop it from working, whereas they are not going to be likely to design their own circuits and add their own custom pinio functions (and in order to do that, they will have to do custom resource assignments as well anyway, so we'd have to do this as well).
As for the tinkerers that come up with brilliant ideas as you describe above, and build their own custom peripherals, I think for users at this technical level the CLI will not be a showstopper, so expecting them to configure their inventions in CLI is not an outlandish ask.